packages/core/admin/admin/src/components/NpsSurvey/index.js

```diff
@@ -123,13 +190,121 @@ const NpsSurvey = () => {
-      <Button onClick={handleSubmitResponse}>Submit</Button>
-    </Flex>
+    <Portal>
+      <Formik
+        initialValues={{ feedback: '', selectedRating: null }}
+        onSubmit={handleSubmitResponse}
+        validationSchema={yup.object({
+          feedback: yup.string(),
+          selectedRating: yup.number(),
+        })}
+      >
+        {({ handleSubmit, values, handleChange, setFieldValue }) => (
+          <Form name="npsSurveyForm" noValidate onSubmit={handleSubmit}>
+            <BannerWrapper hasRadius direction="column" padding={4}>
+              {isFeedbackResponse ? (
+                <Typography fontWeight="semiBold">
+                  {formatMessage({
+                    id: 'app.components.NpsSurvey.feedback-response',
+                    defaultMessage: 'Thank you very much for your feedback!',
+                  })}
+                </Typography>
+              ) : (
+                <>
+                  <Flex justifyContent="space-between" width="100%">
+                    <Header>
+                      <Typography fontWeight="semiBold">
+                        {formatMessage({
+                          id: 'app.components.NpsSurvey.banner-title',
+                          defaultMessage:
+                            'How likely are you to recommend Strapi to a friend or colleague?',
+                        })}
+                      </Typography>
+                    </Header>
+                    <IconButton
+                      onClick={handleDismiss}
+                      aria-label={formatMessage({
+                        id: 'app.components.NpsSurvey.dismiss-survey-label',
+                        defaultMessage: 'Dismiss survey',
+                      })}
+                      icon={<Cross />}
+                    />
+                  </Flex>
+                  <Flex gap={2} marginLeft={8} marginRight={8} marginTop={2} marginBottom={2}>
+                    <Typography variant="pi" textColor="neutral600">
+                      {formatMessage({
+                        id: 'app.components.NpsSurvey.no-recommendation',
+                        defaultMessage: 'Not at all likely',
+                      })}
+                    </Typography>
+                    {ratingArray.map((number) => {
+                      return (
+                        <FieldWrapper
+                          key={number}
+                          className={values.selectedRating === number ? 'selected' : null}
+                        >
+                          <FieldLabel htmlFor={number} id={`${number}-rating`}>
+                            <VisuallyHidden>
+                              <FieldInput
+                                type="radio"
+                                id={number}
+                                name="selectedRating"
+                                checked={values.selectedRating === number}
+                                onChange={() => setFieldValue('selectedRating', number)}
```

Comment:
no it updates the value as string, we need number instead

Reply:
Could we write it e.g. as `onChange={(event) => handleChange(parseInt(event.target.value, 10))}` then?

It makes more sense in case we want to split up the fields at some point ...

Reply:
yes I thought so but this was recommended, https://github.com/jaredpalmer/formik/issues/1191

Reply:
Mh, where do you see that? What they recommend instead:

```
onChange={(e) => {
    setFieldValue(
      "FieldName",
      parseInt(e.target.value)
    );
  }}
```

this would work for me too :)

Reply:
Also: I've checked the design-system: https://github.com/strapi/design-system/blob/main/packages/strapi-design-system/src/NumberInput/NumberInput.tsx#L43C25-L43C25 Could you use `onValueChanged` instead? It should give you a number according to the TS docs.

Reply:
> Mh, where do you see that? What they recommend instead:
> 
> ```
> onChange={(e) => {
>     setFieldValue(
>       "FieldName",
>       parseInt(e.target.value)
>     );
>   }}
> ```
> 
> this would work for me too :)

I meant using "setFieldValue()" method to set the value you need, and I used number directly instead of target value which needs parsing.

Reply:
I see what you mean. I think Fernando's suggestion makes sense, because if we want to split up this component and e.g. have a separate component for each rating item, it is much easier if a callback is used, because then the component is self-contained.